src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -216,6 +338,19 @@ public static FormulaValue DateDiff(EvalVisitor runner, EvalVisitorContext conte
             }
         }
 
+        internal static bool NeedToConvertToUtc(EvalVisitor runner, DateTime datetime, string unit)
+        {
+            // If datetime is already UTC, no need to apply any conversion
+            // For DateAdd in Days or bigger units, we want to preserve the time
+            return datetime.Kind != DateTimeKind.Utc &&
+                   IsSubdayTimeUnit(unit);
+        }
+
+        private static bool IsSubdayTimeUnit(string unit)
+        {
+            return unit is null or "milliseconds" or "seconds" or "minutes" or "hours";
```

Comment:
>unit is null [](http://example.com/codeflow?start=19&length=12)

Is there any case where this is null? If not, we can remove this check. If yes, doesn't it mean "days" - and in this case, it should *not* be a sub-day unit? It would be good to add a test for that (passing Blank() explicitly to the third parameter of something like DateAdd)

Reply:
there's no need, you're right